src/crush/CrushWrapper.cc

```diff
@@ -803,6 +803,11 @@ int CrushWrapper::move_bucket(CephContext *cct, int id, const map<string,string>
 
 int CrushWrapper::swap_bucket(CephContext *cct, int src, int dst)
 {
+  if (choose_args.size() > 0) {
+    ldout(cct, 1) << "swap_bucket not implemented when choose_args is not empty" << dendl;
+    return -EDOM;
+  }
```

Comment:
i think we can actually support this without much trouble.  we're completely swapping the contents, so we can just swap the choose_args for the buckets too.  and then call adjust for the bucket itself at the end to account for any net bucket weight change.